### PR TITLE
Searched model zoo in NAS space hub

### PR DESCRIPTION
**Pull request by ultmaster**
This PR adds `load_searched_model`, which enables a zoo of famous models obtained with NAS.

For first version, I added 18 models on 4 different spaces in total, including the state-of-the-art MobileNetV3Space.
All the models have been (re-)evaluated with a standard evaluation protocol, and the converted pretrained weights have been uploaded to NNI blob.

To make this happen, I made the following changes:

1. A lot of bug fixes in search space hub.
2. New API: `load_searched_model` and `fixed_arch`. I'm planning to add them to a common base class (e.g., `ModelSpace`) in next step.
3. Move `blob_utils` to `nni/common`, so that the download utilities can be shared between space hub and NAS benchmark.
4. Add `dim=-1` in evaluator, to eliminate a DeprecationWarning.
5. Support `min_depth = 0` in Repeat.

### How to test

CI is yet to be added. I plan to add it when I got the VMSS set up.

Manual test example as follows:

```python
model = MobileNetV3Space.load_searched_model('mobilenetv3-large-100', pretrained=True, download=True)

transform = transforms.Compose([
    transforms.Resize(256),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406],
                         std=[0.229, 0.224, 0.225])
])
dataset = ImageNet(directory, 'val', transform=transform)
dataloader = DataLoader(dataset, batch_size=batch_size, num_workers=num_workers)
model.eval()
with torch.no_grad():
    correct = total = 0
    pbar = tqdm.tqdm(dataloader, desc='Evaluating on ImageNet')
    for inputs, targets in pbar:
        logits = model(inputs)
        _, predict = torch.max(logits, 1)
        correct += (predict == targets).cpu().sum().item()
        total += targets.size(0)
        pbar.set_postfix({'correct': correct, 'total': total, 'acc': correct / total * 100})
print('Overall accuracy (top-1):', correct / total * 100)
```

### CODE REVIEW

File: nni/retiarii/hub/README.md [hunk 1]
```diff
@@ -10,20 +10,78 @@ For further motivations and plans, please see https://github.com/microsoft/nni/i
```
Comment (liuzhe-lz) [inline nni/retiarii/hub/README.md:10]: Wait, this file is source, not doc?

Comment (ultmaster) [reply nni/retiarii/hub/README.md:10]: Space hub is in a preview state and hidden from documentation. I think we'll announce it shortly after 1 or 2 more PRs are merged.

File: nni/common/blob_utils.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        if f is not None:
+            shutil.move(f.name, local_path)
+    finally:
+        if f is not None:
+            f.close()
+            if os.path.exists(f.name):
+                os.remove(f.name)
+
+
+def upload_file(local_path: str, destination_path: str, sas_token: str) -> str:
```
Comment (QuanluZhang) [inline nni/common/blob_utils.py:95]: what is `upload_file` used for? who will use this function?

File: nni/retiarii/hub/README.md [hunk 3]
```diff
@@ -10,20 +10,78 @@ For further motivations and plans, please see https://github.com/microsoft/nni/i
...
+| MobileNetV3Space | cream-481             | ImageNet | 79.078 | Default                      |
+| MobileNetV3Space | cream-604             | ImageNet | 79.92  | Default                      |
+| DARTS            | darts-v2              | CIFAR-10 | 97.37  | Default                      |
+| ShuffleNetSpace  | spos                  | ImageNet | 74.14  | BGR tensor; no normalization |
+
+The pretrained weights listed above are converted with [these scripts](https://github.com/ultmaster/spacehub-conversion).
+Do note that some metrics can be higher / lower than the original report,
+because there could be subtle differences between data preprocessing, operation implementation (e.g., 3rd-party hswish vs ``nn.Hardswish``),
+or even library versions we are using. But most of these errors are acceptable (~0.1%).
+We will retrain these architectures in a reproducible and fair training settings, and update these results when the training is ready.
```
Comment (QuanluZhang) [inline nni/retiarii/hub/README.md:81]: what is the meaning of this sentence? if the training is not reproduced, what is the metric in the above table?

File: nni/retiarii/hub/pytorch/mobilenetv3.py [hunk 4]
```diff
@@ -64,92 +83,582 @@ class MobileNetV3Space(nn.Module):
...
-            # stride = 1, residual connection should be automatically enabled
-            InvertedResidual(oup, oup, exp[1], ks[1], squeeze_and_excite=se_blocks[1], activation_layer=act),
-            InvertedResidual(oup, oup, exp[2], ks[2], squeeze_and_excite=se_blocks[2], activation_layer=act),
-            InvertedResidual(oup, oup, exp[3], ks[3], squeeze_and_excite=se_blocks[3], activation_layer=act)
-        ]
+        def layer_builder(idx):
+            exp = nn.ValueChoice(list(self.expand_ratios), label=f's{stage_idx}_i{idx}_exp')
+            ks = nn.ValueChoice([3, 5, 7], label=f's{stage_idx}_i{idx}_ks')
+            # if SE is true, assign a layer choice to SE
+            se_or_skip = cast(Callable[[nn.MaybeChoice[int], nn.MaybeChoice[int]], nn.Module], partial(
```
Comment (QuanluZhang) [inline nni/retiarii/hub/pytorch/mobilenetv3.py:261]: what is meaning of `MaybeChoice`?

File: nni/common/blob_utils.py (hunk 2 above)
Comment (matluster) [reply nni/common/blob_utils.py:95]: As in doctoring: `For maintainers to add...`

File: nni/retiarii/hub/README.md (hunk 3 above)
Comment (matluster) [reply nni/retiarii/hub/README.md:81]: By converting the checkpoints provided by original author to our format, as in L77.

File: nni/retiarii/hub/pytorch/mobilenetv3.py (hunk 4 above)
Comment (matluster) [reply nni/retiarii/hub/pytorch/mobilenetv3.py:261]: int or ValueChoice[int]

File: nni/retiarii/hub/README.md (hunk 3 above)
Comment (QuanluZhang) [reply nni/retiarii/hub/README.md:81]: I see, in L77, would be better to say: The metric column is obtained by evaluating the checkpoints provided the original author and converted in NNI NAS format [these scripts]

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]
